src/fork-safe-mutex.cpp

```diff
@@ -0,0 +1,60 @@
+
```

Comment:
feels weird to create a whole class for this tiny thing, can be a simple function 0o

Reply:
no real other way to minimize changes in different locations